plugin/manager.go

```diff
@@ -253,19 +253,19 @@ func LookupWithCapability(name, capability string) (Plugin, error) {
 	return nil, ErrInadequateCapability{name, capability}
 }
 
-// StateChanged updates plugin internals using from libcontainerd events.
+// StateChanged updates plugin internals using libcontainerd events.
 func (pm *Manager) StateChanged(id string, e libcontainerd.StateInfo) error {
 	logrus.Debugf("plugin state changed %s %#v", id, e)
 
 	switch e.State {
 	case libcontainerd.StateExit:
-		pm.RLock()
-		p, idOk := pm.plugins[id]
-		pm.RUnlock()
-		if !idOk {
-			return ErrNotFound(id)
-		}
 		if pm.shutdown == true {
```

Comment:
What if `Shutdown()` is called when plugin exits. Not a big issue but seems like a race. btw, how is plugin crashing handled?